engine/src/main/java/org/terasology/rendering/cameras/Camera.java

```diff
@@ -239,4 +242,30 @@ public boolean isReflected() {
     public boolean hasInSight(AABB aabb) {
         return viewFrustum.intersects(aabb);
     }
+    
+    /**
+     * Returns True if the head of the player is underwater. False otherwise.
+     *
+     * Implementations must take in account waves if present.
+     *
+     * @param World Rendering Configuration and the World Provider
+     *
+     * @return True if the head of the player is underwater. False otherwise.
+     */    
+    public boolean isCameraUnderWater(RenderingConfig renderingConfig, WorldProvider worldProvider) {
```

Comment:
Let's rename this method to isUnderWater(), so that we do not get the repetition of camera.isCameraUnderWater(). Also, if we get the config and the provider via injection into the camera we'll be able to avoid passing arguments to this method.